include/caffe/solver.hpp

```diff
@@ -129,6 +129,27 @@ class AdaGradSolver : public SGDSolver<Dtype> {
 };
 
 template <typename Dtype>
+class AdaDeltaSolver : public SGDSolver<Dtype> {
+ public:
+  explicit AdaDeltaSolver(const SolverParameter& param)
+      : SGDSolver<Dtype>(param) { PreSolve(); constructor_sanity_check(); }
+  explicit AdaDeltaSolver(const string& param_file)
+      : SGDSolver<Dtype>(param_file) { PreSolve(); constructor_sanity_check(); }
```

Comment:
`virtual` issue addressed in https://github.com/BVLC/caffe/commit/aedff906f6b990c8d23ef2db646bc94ffb6e3497